### PR TITLE
media-sound/murmur: Migrate live ebuild to cmake

### PR DESCRIPTION
**Pull request by kripton**
Closes: https://bugs.gentoo.org/745387
Package-Manager: Portage-3.0.8, Repoman-3.0.2
Signed-off-by: Jannis Achstetter <kripton@kripserver.net>

Equivalent to  #17468: Convert media-sound/murmur-9999 from qmake to cmake. Drops support for pch but adds support for tests. I was unable to test the systemd-files since all my systems run OpenRC but the server itself is working fine.

### CODE REVIEW

**Comment by kripton** [comment]
@Polynomial-C : Any idea why gentoo-bot adds the label "need assignment" and comments `At least one of the listed packages is maintained entirely by non-GitHub developers!`? According to the Gentoo Bugzilla, you are still a Gentoo-dev, right?

**Comment by kripton** [comment]
Fixed issue found by CI (missing line `RESTRICT="!test? ( test )"`)

File: media-sound/murmur/murmur-9999.ebuild [hunk 1]
```diff
@@ -93,36 +94,40 @@ src_prepare() {
...
+		"-Dg15=OFF"
+		"-Dice=$(usex ice)"
+		"-Dgrpc=OFF"
+		"-DBUILD_TESTING=$(usex test)"
+		"-Dzeroconf=$(usex zeroconf)"
 	)
 
-	eqmake5 main.pro -recursive \
-		CONFIG+="${conf_add[*]}"
+	CMAKE_BUILD_TYPE=$(usex debug Debug Release) \
```
Comment (DarthGandalf) [inline media-sound/murmur/murmur-9999.ebuild:118]: You're ignoring CFLAGS from make.conf, use the default build type instead.

Comment (kripton) [reply media-sound/murmur/murmur-9999.ebuild]: Thanks for the review! I've removed the line after having checked other ebuilds using the `cmake` ECLASS

Comment (DarthGandalf) [reply media-sound/murmur/murmur-9999.ebuild]: But if `debug` USE flag is not used now, remove it from IUSE

**Comment by DarthGandalf** [comment]
You'll need to ping PolynomialC in IRC, btw. I'm just passing by

File: media-sound/murmur/murmur-9999.ebuild (hunk 1 above)
Comment (kripton) [reply media-sound/murmur/murmur-9999.ebuild]: Hm, not so sure about that one... keepassxc also uses the `cmake` eclass, has debug in IUSE but doesn't actually use it inside the ebuild:
https://github.com/gentoo/gentoo/blob/64fe952caa63f4916a5e2f704ebd5306ff00198f/app-admin/keepassxc/keepassxc-2.6.2.ebuild#L27

Comment (kripton) [reply media-sound/murmur/murmur-9999.ebuild]: Hehe, nice one. I thought maybe the cmake eclass would use that info somehow. I've removed the USE flag from my proposed murmur-update now

Comment (DarthGandalf) [reply media-sound/murmur/murmur-9999.ebuild]: Usually if a eclass makes use of a USE flag, it adds it to IUSE itself

File: media-sound/murmur/murmur-9999.ebuild [hunk 2]
```diff
@@ -93,36 +95,39 @@ src_prepare() {
...
+	local mycmakeargs=(
+		"-DBUILD_TESTING=$(usex test)"
+		"-Dclient=OFF"
+		"-Ddbus=$(usex dbus)"
+		"-Dg15=OFF"
+		"-Dgrpc=$(usex grpc)"
+		"-Dice=$(usex ice)"
+		"-Doverlay=OFF"
+		"-Dserver=ON"
+		"-Dzeroconf=$(usex zeroconf)"
```
Comment (Polynomial-C) [inline media-sound/murmur/murmur-9999.ebuild:116]: Please also fix the quoting like I did in the mumble ebuild in commit 3629693f926f2a47c4adc08e28987bb265a7aa4a

**Review by Polynomial-C — CHANGES_REQUESTED** [review]
Looking good. There's only the quoting that needs fixing (see my other comment)

File: media-sound/murmur/murmur-9999.ebuild (hunk 2 above)
Comment (kripton) [reply media-sound/murmur/murmur-9999.ebuild]: > Looking good. There's only the quoting that needs fixing (see my other comment)

Thanks for the review. Quoting has been adapted and ebuild re-tested

**Pull request closed without merging**